src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -56,6 +56,9 @@ public sealed partial class Utf8JsonWriter : IDisposable, IAsyncDisposable
         // else, no list separator is needed since we are writing the first item.
         private int _currentDepth;
 
+        private Memory<byte>? _indentBytes;
+        private byte? _indentByte;
```

Comment:
You could probably avoid making these two fields nullable if you use an empty `Memory<byte>` to indicate that `_indentByte` should be used instead.

Reply:
I have tried to avoid nullable fields but an empty `IndentText` is valid and should be supported here.

Reply:
It's all guarded by the `Indent` flag, so I think it is reasonable to require that the text is non-empty.

Reply:
For now I consider that an empty `IndentText` is allowed (as it is supported by the JSON specification).
However, I have removed the nullability for `Memory<byte>` but kept for the single byte.